back/src/database/SpotSQL.js

```diff
@@ -1,43 +1,64 @@
-            text: 'INSERT INTO spots.review(review_id, spot_id, comment, score, user_id) VALUES($1, $2, $3, $4, $5);',
-            values: [ this.review_id, this.spot_id, this.comment, this.score, this.user_id]
-        };
-        const querys = [query1, query2];
-        conf.connect(querys);
-    }
-
-}
-
-class Review{
-    constructor(review_id, spot_id, comment, score, user_id){
-        this.review_id = review_id;
-        this.spot_id = spot_id;
-        this.comment = comment;
-        this.score = score;
-        this.user_id = user_id;
-    }
+async function saveSpot(newSpot){
+    const query1 = {
+        text: 'INSERT INTO spots.spots(spot_id, spot_name, geom, picture, spot_type, user_id) VALUES($1, $2, $3, $4, $5, $6);',
+        values: [newSpot.spot_id, newSpot.spot_name, newSpot.geom, newSpot.picture, newSpot.spot_type, newSpot.user_id]
+    };
+    const query2 = {
+        text: 'INSERT INTO spots.review(review_id, spot_id, comment, score, user_id) VALUES($1, $2, $3, $4, $5);',
+        values: [newSpot.review_id, newSpot.spot_id, newSpot.comment, newSpot.score, newSpot.user_id]
+    };
+    const deleteSpotQuery = {
+        text: 'DELETE from spots.spots where spot_id = $1;',
+        values: [newSpot.spot_id]
+    };
+    connection.connect().then(()=>{
+        //TODO: NULL CHECK
+        return connection.query(query1)
+        .then(()=>{
+            return connection.query(query2)
+            .then(()=>{
+                connection.end()
+                .then(()=>{
+                    debug(fileLabel,"saved spot: " + newSpot);
+                    return {"sucess":true,"data":newSpot};
+                })
+            })
+            .catch((exception)=>{
+                error(fileLabel,"Error while saving review: " + exception);
+                return connection.query(deleteSpotQuery)
+                .then(()=>{
+                    connection.end()
+                    .then(()=> {
+                        debug(fileLabel,"deleted spot: " + newSpot);
+                        return {"sucess":true,"data":newSpot};
+                    });
+                })
+                .catch(()=>{
+                    error(fileLabel,"Error while deleting spot: " + exception);
+                    return {"success":false,"data":exception};
+                })
+            });
+        })
+        .catch((exception)=>{
+            connection.end().then(()=>{
+                error(fileLabel,"Error while saving spot: " + exception);
+                return {"success":false,"data":exception};
+            });
+        });
+    }).catch((exception)=>{
+        error(fileLabel,"Error trying to connect to database: " + exception);
+        return {"sucess":false,"data":exception};
+    });
 }
 
+const testSpot = new Spot(4311, 'spotname3', null, 'picture', 'spottype3', "user_id111", 67, 'comment2', 90);
+saveSpot(testSpot);
 // new Spot(8, 'spotname3', null, 'picture', 'spottype3', 3, 6, 'comment2', 90).addToDatabase();
```

Comment:
You need to add 
module.exports={
Function1OutsideFile:Function1insideFile
,Function2:Function2,
...} at the end otherwise we cannot call any of the functions in this file. For example if you want to export function saveSpot() as saveSpot, you need to add 
module.exports = {saveSpot:saveSpot};